src/inputs/ProjectInput.ts

```diff
@@ -1,29 +1,52 @@
 import { InputType, Field, ID } from 'type-graphql';
+import {
+  MaxLength,
+  IsNotEmpty,
+  IsString,
+  IsMongoId,
+  IsIn,
+} from 'class-validator';
 
 @InputType()
 export default class ProjectInput {
   @Field(() => ID)
+  // @jf.string().required()
+  @IsNotEmpty()
   created_by: string;
 
   @Field()
+  @IsNotEmpty()
+  @IsString()
+  @MaxLength(30, {
+    message: 'Project Name must be between 1 and 30 characters',
+  })
   name: string;
 
   @Field()
+  @IsNotEmpty()
+  @IsString()
+  @MaxLength(30, {
+    message: 'Project Name must be between 1 and 30 characters',
+  })
+  // contains specified values ?
```

Comment:
Yes. Statuses to be confirmed, but they should also be in an enum file (as you did for tickets)